### ISSUE TITLE
Clarify Semantics

### ISSUE THREAD

**Issue by tcurdt** [mustafaturan/bus#5]
From the readme it's unclear how events are consumed. Will events always make it to all consumers? or can a consumer stop the processing? What is the order of processing of events? Is the order guaranteed? Would be great to clarify this.

**Comment by mustafaturan** [comment]
Hey @tcurdt 

TLDR; Everything is synchronously until the Handler receive the event. The Handler decides if it needs process async or not. 

1. Events always delivers synchronously to the consumers that match the topic name. Topic name matching is done by Handler's Regex pattern. **Ref:** https://godoc.org/github.com/mustafaturan/bus#Bus.RegisterHandler

2. The order of processing events up to the Handler, it can consume any time after the consumer received. 

3. The order of delivery to events guaranteed per event but not per `Handler`. 
**Ref:** https://github.com/mustafaturan/bus/blob/master/bus.go#L227

4. Consumer can stop receiving the by deregistering itself. **Ref:** https://godoc.org/github.com/mustafaturan/bus#Bus.DeregisterHandler

**Comment by tcurdt** [comment]
Thanks! Would be a good addition to the README.
